### PR TITLE
MAKE-1274: Implement HistoricalTestData SaveNew method

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: model/historical_test_data.go [hunk 1]
```diff
@@ -101,6 +102,37 @@ func (d *HistoricalTestData) Find(ctx context.Context) error {
...
+		return errors.New("cannot save unpopulated historical test data")
+	}
+	if d.env == nil {
+		return errors.New("cannot find with a nil environment")
+	}
+
+	data, err := bson.Marshal(d)
+	if err != nil {
+		return errors.Wrap(err, "problem marshalling historical test data")
+	}
```
Comment (tychoish) [inline model/historical_test_data.go:116]: I'd do this on line 128? We might also want to wrap this in something that returns a reader? 

Do we expect that we'll have more than one document at a time? Should we have some way to batch this? 

Also does it make sense to use bson or json for the historical rollups?

Comment (julianedwards) [reply model/historical_test_data.go:116]: I thought we decided on bson? Does it matter?

Comment (tychoish) [reply model/historical_test_data.go:116]: I think BSON for the raw data, but maybe JSON for the rollups? eh, bson is fine.

Comment (julianedwards) [reply model/historical_test_data.go:116]: It would be one document at a time, these will be created and updated in amboy jobs as results come in.

**Pull request merged**